packages/salesforcedx-vscode-lwc-mobile/src/commands/forceLightningLwcMobile.ts

```diff
@@ -0,0 +1,212 @@
+    return;
+  }
+
+  if (!fs.existsSync(resourcePath)) {
+    const message = nls.localize(
+      'force_lightning_lwc_file_nonexist',
+      resourcePath
+    );
+    showError(new Error(message));
+    return;
+  }
+
+  const isSFDX = true; // TODO support non SFDX projects
+  const isDirectory = fs.lstatSync(resourcePath).isDirectory();
+  const componentName = isDirectory
+    ? componentUtil.moduleFromDirectory(resourcePath, isSFDX)
+    : componentUtil.moduleFromFile(resourcePath, isSFDX);
+  if (!componentName) {
+    const message = nls.localize(
+      'force_lightning_lwc_preview_unsupported',
+      resourcePath
+    );
+    showError(new Error(message));
+    return;
+  }
+
+  const platformSelection = await vscode.window.showQuickPick(platformInput, {
+    placeHolder: nls.localize('force_lightning_lwc_mobile_platform_selection')
+  });
+  if (!platformSelection) {
+    vscode.window.showWarningMessage(
+      nls.localize('force_lightning_lwc_mobile_cancelled')
+    );
+    return;
+  }
+
+  let target: string = platformSelection.defaultTargetName;
+  let placeholderText = nls.localize(
+    'force_lightning_lwc_mobile_target_default'
+  );
+  const rememberDeviceConfigured =
+    getWorkspaceSettings().get('rememberDevice') || false;
+  const lastTarget = getRememberedDevice(platformSelection);
+
+  // Remember device setting enabled and previous device retrieved.
+  if (rememberDeviceConfigured && lastTarget) {
+    placeholderText = nls.localize(
+      'force_lightning_lwc_mobile_target_remembered',
+      lastTarget
+    );
+    target = lastTarget;
+  }
+  const targetName = await vscode.window.showInputBox({
+    placeHolder: placeholderText
+  });
+
+  if (targetName === undefined) {
+    vscode.window.showInformationMessage(
+      nls.localize('force_lightning_lwc_mobile_device_cancelled')
+    );
+    return;
+  }
+
+  // New target device entered
+  if (targetName !== '') {
+    updateRememberedDevice(platformSelection, targetName);
+    target = targetName;
+  }
+
+  const mobileCancellationTokenSource = new vscode.CancellationTokenSource();
+  const mobileCancellationToken = mobileCancellationTokenSource.token;
+
+  // TODO: Remove `fullUrl` and use `resourcePath` when SFDX plugin is ready.
+  const fullUrl = `http://localhost:3333/lwc/preview/${componentName}`;
+  // TODO: Add setting for loglevel
+  const command = new SfdxCommandBuilder()
+    .withDescription(commandName)
+    .withArg('force:lightning:local:preview')
+    .withFlag('-p', platformSelection.platformName)
+    .withFlag('-t', target || platformSelection.defaultTargetName)
+    .withFlag('-d', fullUrl)
+    .build();
+
+  const mobileExecutor = new CliCommandExecutor(command, {
+    env: { SFDX_JSON_TO_STDOUT: 'true' }
+  });
+  const execution = mobileExecutor.execute(mobileCancellationToken);
+  telemetryService.sendCommandEvent(logName, startTime);
+
+  execution.processExitSubject.subscribe(async exitCode => {
```

Comment:
FYI,  I'm assuming that the problem may stem from relying on just exit events?

Reply:
@trooper2013 Circling back on this.  Works perfectly with iOS runs and for detecting if the SFDX plugin is not installed.  However, it does not work well with Android because the process does not end until the command/emulator is killed.  Is the long term plan to spawn the emulator in a separate process and end with success or failure like iOS?

Reply:
And how does that work with eventual server management?